### PR TITLE
Changes in typed signature validation and normalization

### PR DESCRIPTION
**Pull request by jpuri**
The PR addresses these 2 issues:

https://github.com/MetaMask/metamask-extension/issues/25733
https://github.com/MetaMask/MetaMask-planning/issues/2810

### CODE REVIEW

File: src/utils/normalize.ts [hunk 1]
```diff
@@ -54,36 +52,14 @@ function parseTypedMessage(data: string) {
...
  * @param address - The address to normalize.
  * @returns The normalized address.
  */
-function normalizeContractAddress(address: string): Hex | string {
-  if (isStrictHexString(address) && isValidHexAddress(address)) {
-    return address;
+function normalizeContractAddress(address: Hex): Hex {
+  if (address.startsWith('0X')) {
+    return `0x${address.slice(2)}`;
   }
```
Comment (jpuri) [inline src/utils/normalize.ts:63]: Normilizing decimal and octal value is removed now, only normalization now is replacing `0X` prefix with `0x`.

**Review by mcmire — DISMISSED** [review]
LGTM!

**Comment by mcmire** [comment]
Oof, I see that CI is failing. This could be either a problem where the cache is not being properly restored, or the incorrect set of files is being saved in the cache. In other repos we've discussed replacing `yarn --immutable --immutable-cache` with `yarn --immutable` in `.github/workflows/build-lint-test.yml`. I'll try to investigate soon, but you could experimenting with that if you're stuck.

**Comment by cryptodev-2s** [comment]
> > Oof, I see that CI is failing. This could be either a problem where the cache is not being properly restored, or the incorrect set of files is being saved in the cache. In other repos we've discussed replacing `yarn --immutable --immutable-cache` with `yarn --immutable` in `.github/workflows/build-lint-test.yml`. I'll try to investigate soon, but you could experimenting with that if you're stuck.
> 
> Saw this in other repos when rerunning previously passing jobs. Also specifically Node 22.
> 
> Node.js 22.5.0 was just released so maybe something [lurking here](https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V22.md#22.5.0)?
> 
> Unless it's simply an intermittent infra issue at GH.



> > Oof, I see that CI is failing. This could be either a problem where the cache is not being properly restored, or the incorrect set of files is being saved in the cache. In other repos we've discussed replacing `yarn --immutable --immutable-cache` with `yarn --immutable` in `.github/workflows/build-lint-test.yml`. I'll try to investigate soon, but you could experimenting with that if you're stuck.
> 
> Saw this in other repos when rerunning previously passing jobs. Also specifically Node 22.
> 
> Node.js 22.5.0 was just released so maybe something [lurking here](https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V22.md#22.5.0)?
> 
> Unless it's simply an intermittent infra issue at GH.

@legobeat a fix for this has just been included in a new node patch [`v22.5.1`](https://github.com/nodejs/node/pull/53935)

**Comment by mcmire** [comment]
I've fixed the Node issue here: https://github.com/MetaMask/eth-json-rpc-middleware/pull/321

**Review by matthewwalsh0 — DISMISSED** [review]

**Review by OGPoyraz — APPROVED** [review]

**Review by mcmire — DISMISSED** [review]
Looks good!

File: .github/workflows/build-lint-test.yml [hunk 2]
```diff
@@ -32,7 +32,7 @@ jobs:
         with:
           node-version: ${{ matrix.node-version }}
           cache: 'yarn'
-      - run: yarn --immutable --immutable-cache
```
Comment (cryptodev-2s) [inline .github/workflows/build-lint-test.yml:35]: Can we rollback this change ?

**Review by cryptodev-2s — CHANGES_REQUESTED** [review]

File: .github/workflows/build-lint-test.yml (hunk 2 above)
Comment (jpuri) [reply .github/workflows/build-lint-test.yml]: 👍 I updated the PR

**Review by cryptodev-2s — APPROVED** [review]
LGTM!

**Pull request merged**